### PR TITLE
Config network

### PR DESCRIPTION
**Pull request by zelfroster**
Added features to set Dynamic DNS, configure TOR/I2P SOCKS Proxy from the webui.
Fixed some UI issues and refactored code.

Some issues are still there but not from webui, they need to be fixed/implemented on the libretroshare, probably.

### CODE REVIEW

**Comment by zelfroster** [comment]
Here is what it looks like from the webui.
1686047390_Jun06_15:59:50

**Comment by zelfroster** [comment]
Now, Everything is working perfectly. This PR is also ready for review.
1686334902_Jun09_23:51:42

File: webui-src/app/config/config_util.js [hunk 1]
```diff
@@ -1,25 +1,65 @@
...
+/* NAT Net Mode */
+const RS_NETMODE_UDP = 1;
+const RS_NETMODE_UPNP = 2;
+const RS_NETMODE_EXT = 3;
+
+function getRandomId(tagArr) {
+  const random = Math.floor(Math.random() * (MAX_TAG_ID_VAL - MIN_TAG_ID_VAL) + MIN_TAG_ID_VAL);
+  tagArr.forEach((tag) => {
+    if (tag.key === random) {
+      return getRandomId();
```
Comment (rottencandy) [inline webui-src/app/config/config_util.js:36]: calling without array would cause an error I think

Comment (zelfroster) [reply webui-src/app/config/config_util.js]: Sorry, can't believe I forgot this 😅. Fixed it.

**Review by rottencandy — APPROVED** [review]
Looks good, thanks

**Comment by zelfroster** [comment]
> Looks good, thanks

Thanks to you, And I had a question about mithril.js.
In `config_network.js`, line number 366, in main component `Component`.
We have two variables `sslId` and `component` and we are using `oninit` to fetch the data and store in the variables. So, I tried passing `sslId` and `details` to the other components as props from the main component, which were doing the same fetch call for `sslId` and `details` inside them using `oninit`.
But after fetching the data in the main Component, they didn't seem to update the props (variables). Since, after fetching only the `view` UI is redrawn, so I tried updating them in the `view`, but that resulted in an infinite re-rendering.

After researching, I found that, `m.stream()` could be used to have a `state` like functionality like what we have in React. But I think the version of mithril.js used in webui doesn't have m.stream() functionality.

Is there any way to do it, so we could make the code more reusable and reduce the number of data fetching? @rottencandy

**Comment by zelfroster** [comment]
> You could try promise-chaining the requests so that there are no nested requests. If that doesn't work you could try manually triggering a rerender with m.redraw.

I had already tried promise-chaining + manual rerender with m.redraw(). But It didn't work.

**Pull request merged**